src/codegen.cpp

```diff
@@ -4376,6 +4376,22 @@ static void emit_varinfo_assign(jl_codectx_t &ctx, jl_varinfo_t &vi, jl_cgval_t
     }
 }
 
+static void emit_binding_store(jl_codectx_t &ctx, jl_binding_t *bnd, Value *bp, jl_value_t *r, ssize_t ssaval, AtomicOrdering Order)
+{
+    assert(bnd);
+    Value *rval = boxed(ctx, emit_expr(ctx, r, ssaval));
+    if (!bnd->constp && bnd->ty && jl_isa(r, bnd->ty)) {
+        StoreInst *v = ctx.builder.CreateAlignedStore(rval, bp, Align(sizeof(void*)));
+        v->setOrdering(Order);
+        //tbaa_decorate(ctx.tbaa().tbaa_binding, v);
```

Comment:
```suggestion
        tbaa_decorate(ctx.tbaa().tbaa_binding, v);
```
(not needed for correctness, but good to have)

Reply:
```suggestion
        //tbaa_decorate(ctx.tbaa().tbaa_binding, v);
        ctx.builder.CreateCall(prepare_call(jl_write_barrier_binding_func), {v});
```

You will also need to implement `jl_write_barrier_binding_func` by cloning the way `jl_write_barrier_func` works, and changing the function it calls inside (c.f. `jl_gc_wb_binding` vs `jl_gc_wb`)

Reply:
I tried implementing `jl_write_barrier_binding_func` here, but I still get segfaults, so I probably missed something. I must admit that I'm a bit out of my depth here, do you see what the problem here could be?